app/models/user.rb

```diff
@@ -105,7 +105,8 @@ def build_from_invitation(invitation)
     self.name = invitation.name
     self.email = invitation.email
     skip_confirmation!
-    course_users.build(course: invitation.course, name: invitation.name,
+    course_role = invitation.role || Course::CourseUser.roles[:student]
```

Comment:
Can reference from this discussion https://github.com/Coursemology/coursemology2/pull/2468#discussion_r135431776

> I'd prefer if we kept this default convention in the `course_users` file, instead of having it in the `user` model. In the future when we change defaults, it would be easier to locate these defaults. Maybe you can consider modifying the `set_defaults` method within the `course_user` model, or if you have a better method, you can do that!

Reply:
Ah ok, I think I set the defaults wrongly in my previous attempts and caused a bunch of null violations in the database. Will try again since this method you suggest is way cleaner.